### DISCUSSION TITLE
Question about 10.0.0.x IPs

### DISCUSSION THREAD

**Discussion by dain7672** [ANLAB-KAIST/KENSv3#102] [Project Q&A 1. Application Layer (Socket)] [answered]
Hello.
I have a question about 10.0.0.x IP addresses.

Currently, when I try below, it works without any errors.
`echo-non-kens c 127.0.0.1 9000 echo-test`
`echo-non-kens s 0.0.0.0 9000 server-hello`

But when I change 127.0.0.1 -> 10.0.0.1 and execute echo-non-kens again, timeout occurs.
(10.0.0.1 is an IP used in echo test case)
So I guess that 10.0.0.1 is a special IP only used in KENS that routes to 127.0.0.1. Is it right?

In addition,
Although echo-non-kens works, when I execute the same code as `echo`, Connect and Accept just fail with errno: 0.
What can I do to fix it?

**Comment by whkim0** [comment, marked as answer]
Yes, KENS in `echo` hosts echo servers with IP addresses of `10.0.0.x` and updates the routing table. `echo-non-kens` runs without KENS, thus you cannot expect any connection with `10.x.x.x`. They are reserved for private networks. (https://datatracker.ietf.org/doc/html/rfc1918#section-3)

Have you tried using a different port number? Also, KENS does not support errno. https://github.com/ANLAB-KAIST/KENSv3/discussions/43#discussioncomment-2445441

**Reply by dain7672** [reply]
Thanks!
